src/Compilers/CSharp/Test/Semantic/Semantics/RecordTests.cs

```diff
@@ -3655,5 +3658,645 @@ private static ImmutableArray<Symbol> GetProperties(CSharpCompilation comp, stri
+                Diagnostic(ErrorCode.ERR_UnexpectedArgumentList, "(").WithLocation(17, 24)
+                );
+
+            var tree = comp.SyntaxTrees.First();
+            var model = comp.GetSemanticModel(tree);
+
+            var x = tree.GetRoot().DescendantNodes().OfType<IdentifierNameSyntax>().Where(id => id.Identifier.ValueText == "X").First();
+            Assert.Equal("Base(X, Y)", x.Parent!.Parent!.Parent!.ToString());
+
+            var symbolInfo = model.GetSymbolInfo(x);
+            Assert.Null(symbolInfo.Symbol);
+            Assert.Empty(symbolInfo.CandidateSymbols);
+            Assert.Equal(CandidateReason.None, symbolInfo.CandidateReason);
+            Assert.Same("<global namespace>", model.GetEnclosingSymbol(x.SpanStart).ToTestDisplayString());
+            Assert.Empty(model.LookupSymbols(x.SpanStart, name: "X"));
+            Assert.DoesNotContain("X", model.LookupNames(x.SpanStart));
+        }
+
+        [Fact]
+        public void BaseArguments_05()
+        {
+            var src = @"
+using System;
+
+record Base
+{
+    public Base(int X, int Y)
+    {
+    }
+
+    public Base() {}
+}
+
+partial record C : Base(X, Y)
+{
+}
+
+partial record C : Base(X, Y)
+{
+}
+";
+
+            var comp = CreateCompilation(src);
+            comp.VerifyEmitDiagnostics(
+                // (13,24): error CS8861: Unexpected argument list.
+                // partial record C : Base(X, Y)
+                Diagnostic(ErrorCode.ERR_UnexpectedArgumentList, "(").WithLocation(13, 24),
+                // (17,24): error CS8861: Unexpected argument list.
+                // partial record C : Base(X, Y)
+                Diagnostic(ErrorCode.ERR_UnexpectedArgumentList, "(").WithLocation(17, 24)
+                );
+
+            var tree = comp.SyntaxTrees.First();
+            var model = comp.GetSemanticModel(tree);
+
+            var xs = tree.GetRoot().DescendantNodes().OfType<IdentifierNameSyntax>().Where(id => id.Identifier.ValueText == "X").ToArray();
+            Assert.Equal(2, xs.Length);
+
+            foreach (var x in xs)
+            {
+                Assert.Equal("Base(X, Y)", x.Parent!.Parent!.Parent!.ToString());
+
+                var symbolInfo = model.GetSymbolInfo(x);
+                Assert.Null(symbolInfo.Symbol);
+                Assert.Empty(symbolInfo.CandidateSymbols);
+                Assert.Equal(CandidateReason.None, symbolInfo.CandidateReason);
+                Assert.Same("<global namespace>", model.GetEnclosingSymbol(x.SpanStart).ToTestDisplayString());
+                Assert.Empty(model.LookupSymbols(x.SpanStart, name: "X"));
+                Assert.DoesNotContain("X", model.LookupNames(x.SpanStart));
+            }
+        }
+
+        [Fact]
+        public void BaseArguments_06()
+        {
+            var src = @"
+using System;
+
+record Base
+{
+    public Base(int X, int Y)
+    {
+    }
+
+    public Base() {}
+}
+
+partial record C(int X, int Y) : Base(X, Y)
```

Comment:
I assume the following is legal:

```C#
record Base(int X);
partial record C(int X) : Base(X);
partial record C { }
```

Could you write a test that shows it succeeds/emits correctly?

Reply:
> I assume the following is legal:

Yes, this works, adding test in the next PR

---
In reply to: [436186239](https://github.com/dotnet/roslyn/pull/44876#discussion_r436186239) [](ancestors = 436186239)

Reply:
> adding test in the next PR

Added in [https://github.com/dotnet/roslyn/pull/44906](https://github.com/dotnet/roslyn/pull/44906)

---
In reply to: [436197211](https://github.com/dotnet/roslyn/pull/44876#discussion_r436197211) [](ancestors = 436197211,436186239)